hooks/contractHooks.ts

```diff
@@ -40,7 +40,7 @@ export const useContractLoader = <
   useEffect(() => {
     // noinspection JSIgnoredPromiseFromCall
     fetchValues();
-  }, [contract]);
+  }, [address]);
```

Comment:
these probably need wallet as well. However, I wonder if taking away factory from line 36 could fix it as well

Reply:
I've confirmed that instead of changing this from contract to address, removing factory in the useMemo expression fixes it, and contract gets memoized properly. The problem is really that factory should have been memoized

Reply:
done!